<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>miaSim: Microbiome Data Simulation • miaSim</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="miaSim: Microbiome Data Simulation">
<meta property="og:description" content="miaSim">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">miaSim</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/miaSim.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/microbiome/miaSim/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="miaSim_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>miaSim: Microbiome Data Simulation</h1>
                        <h4 data-toc-skip class="author">Yagmur Simsek</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:yagmur.simsek@hsrw.org" class="email">yagmur.simsek@hsrw.org</a>
      
                              <h4 data-toc-skip class="author">Yu Gao</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:gaoyu19920914@gmail.com" class="email">gaoyu19920914@gmail.com</a>
      
                              <h4 data-toc-skip class="author">Emma Gheysen</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:emma.gheysen@student.kuleuven.be" class="email">emma.gheysen@student.kuleuven.be</a>
      
                              <h4 data-toc-skip class="author">Daniel Rios Garza</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:danielrios.garza@kuleuven.be" class="email">danielrios.garza@kuleuven.be</a>
      
                              <h4 data-toc-skip class="author">Karoline Faust</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:karoline.faust@kuleuven.be" class="email">karoline.faust@kuleuven.be</a>
      
                              <h4 data-toc-skip class="author">Leo Lahti</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:leo.lahti@iki.fi" class="email">leo.lahti@iki.fi</a>
      
                  
            <h4 data-toc-skip class="date">2022-07-23</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/microbiome/miaSim/blob/HEAD/vignettes/miaSim.Rmd" class="external-link"><code>vignettes/miaSim.Rmd</code></a></small>
      <div class="hidden name"><code>miaSim.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>miaSim</code> implements tools for microbiome data simulation based on different ecological modeling assumptions. These can be used to simulate species abundance matrices, including time series. For a detailed function documentation, see the <a href="https://microbiome.github.io/miaSim/reference/index.html" class="external-link">function reference page</a></p>
<p>Install the Bioconductor release version with</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span></code></pre></div>
<p>Load the library</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/miaSim" class="external-link">miaSim</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h3>
<div class="section level4">
<h4 id="generating-species-interaction-matrices">Generating species interaction matrices<a class="anchor" aria-label="anchor" href="#generating-species-interaction-matrices"></a>
</h4>
<p>Some of the models rely on interaction matrices that represents interaction heterogeneity between species. The interaction matrix can be generated with different distributional assumptions.</p>
<p>Generate interactions from normal distribution:</p>
<p>Generate interactions from uniform distribution:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">A_uniform</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/randomA.html">randomA</a></span><span class="op">(</span>n_species <span class="op">=</span> <span class="fl">10</span>, d <span class="op">=</span> <span class="op">-</span><span class="fl">0.4</span>, min_strength <span class="op">=</span> <span class="op">-</span><span class="fl">0.8</span>,</span>
<span>                        max_strength <span class="op">=</span> <span class="fl">0.8</span>, connectance <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="generalized-lotka-volterra-glv">Generalized Lotka-Volterra (gLV)<a class="anchor" aria-label="anchor" href="#generalized-lotka-volterra-glv"></a>
</h4>
<p>The generalized Lotka-Volterra simulation model generates time-series assuming microbial population dynamics and interaction.</p>
</div>
<div class="section level4">
<h4 id="ricker-model">Ricker model<a class="anchor" aria-label="anchor" href="#ricker-model"></a>
</h4>
<p>Ricker model is a discrete version of the gLV:</p>
<p>The number of species specified in the interaction matrix must be the same as the species used in the models.</p>
</div>
<div class="section level4">
<h4 id="hubbell-model">Hubbell model<a class="anchor" aria-label="anchor" href="#hubbell-model"></a>
</h4>
<p>Hubbell Neutral simulation model characterizes diversity and relative abundance of species in ecological communities assuming migration, births and deaths but no interactions. Losses become replaced by migration or birth.</p>
<p>One can also simulate parameters for the Hubbell model.</p>
</div>
<div class="section level4">
<h4 id="self-organised-instability-soi">Self-Organised Instability (SOI)<a class="anchor" aria-label="anchor" href="#self-organised-instability-soi"></a>
</h4>
<p>The Self-Organised Instability (SOI) model generates time series for communities and accelerates stochastic simulation.</p>
</div>
<div class="section level4">
<h4 id="stochastic-logistic-model">Stochastic logistic model<a class="anchor" aria-label="anchor" href="#stochastic-logistic-model"></a>
</h4>
<p>Stochastic logistic model is used to determine dead and alive counts in community.</p>
</div>
<div class="section level4">
<h4 id="consumer-resource-model">Consumer-resource model<a class="anchor" aria-label="anchor" href="#consumer-resource-model"></a>
</h4>
<p>The consumer resource model requires the use of the <code>randomE</code> function, which returns a matrix containing the production rates and consumption rates of each species. The resulting matrix is used as a determination of resource consumption efficiency.</p>
</div>
</div>
<div class="section level3">
<h3 id="data-containers">Data containers<a class="anchor" aria-label="anchor" href="#data-containers"></a>
</h3>
<p>The abundance matrices can be converted into <code>TreeSummarizedExperiment</code> object. This provides access to a broad range of tools for microbiome analysis that support this format (see <a href="http://microbiome.github.io" class="external-link">microbiome.github.io</a>). More examples on <code>TreeSummarizedExperiment</code> object manipulation and analysis can be found at <a href="https://microbiome.github.io/OMA" class="external-link">OMA Online Manual</a>.</p>
<p>This example shows how to generate taxonomic abundance data, including sample metadata, and convert this into into <code>TreeSummarizedExperiment</code>.</p>
<p>The community simulations result in abundance matrix that can be stored in <code>TreeSummarizedExperiment</code> <span class="citation">[@TreeSE]</span> class object. Other fields, such as rowData containing information about the samples, and colData, consisting of sample metadata describing the samples, can be added to the <code>TreeSummarizedExperiment</code> class object.</p>
<p>A broad range of functions are available in other packages to explore data in the <code>TreeSummarizedExperiment</code> format (see e.g. microbiome.github.io).</p>
<p>For instance, to plot population density we can use the <code>miaViz</code> package:</p>
<div class="section level4">
<h4 id="auxiliary-functions">Auxiliary functions<a class="anchor" aria-label="anchor" href="#auxiliary-functions"></a>
</h4>
<p>The <code>simulationTimes</code> function generates lists of time series that can be specified as simulation time and time points to keep in simulated time.</p>
<p>It is also possible to incorporate phylogenetic tree information into <code>TreeSummarizedExperiment</code> objects. For further details, see:</p>
</div>
</div>
<div class="section level3">
<h3 id="related-work">Related work<a class="anchor" aria-label="anchor" href="#related-work"></a>
</h3>
<ul>
<li>
<a href="https://github.com/OSS-Lab/micodymora" class="external-link">micodymora</a> Python package for microbiome simulation</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /home/xxx/bin/R-4.2.0/lib/libRblas.so</span></span>
<span><span class="co">## LAPACK: /home/xxx/bin/R-4.2.0/lib/libRlapack.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] miaSim_1.3.1                   TreeSummarizedExperiment_2.5.0</span></span>
<span><span class="co">##  [3] Biostrings_2.65.1              XVector_0.37.0                </span></span>
<span><span class="co">##  [5] SingleCellExperiment_1.19.0    SummarizedExperiment_1.27.1   </span></span>
<span><span class="co">##  [7] Biobase_2.57.1                 GenomicRanges_1.49.0          </span></span>
<span><span class="co">##  [9] GenomeInfoDb_1.33.3            IRanges_2.31.0                </span></span>
<span><span class="co">## [11] S4Vectors_0.35.1               BiocGenerics_0.43.0           </span></span>
<span><span class="co">## [13] MatrixGenerics_1.9.1           matrixStats_0.62.0            </span></span>
<span><span class="co">## [15] BiocStyle_2.25.0              </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] sass_0.4.1             tidyr_1.2.0            jsonlite_1.8.0        </span></span>
<span><span class="co">##  [4] gtools_3.9.2.2         bslib_0.3.1            assertthat_0.2.1      </span></span>
<span><span class="co">##  [7] BiocManager_1.30.18    yulab.utils_0.0.5      GenomeInfoDbData_1.2.8</span></span>
<span><span class="co">## [10] yaml_2.3.5             pillar_1.8.0           lattice_0.20-45       </span></span>
<span><span class="co">## [13] glue_1.6.2             digest_0.6.29          htmltools_0.5.3       </span></span>
<span><span class="co">## [16] Matrix_1.4-1           pkgconfig_2.0.3        bookdown_0.27         </span></span>
<span><span class="co">## [19] zlibbioc_1.43.0        purrr_0.3.4            tidytree_0.3.9        </span></span>
<span><span class="co">## [22] pracma_2.3.8           BiocParallel_1.31.10   tibble_3.1.8          </span></span>
<span><span class="co">## [25] generics_0.1.3         cachem_1.0.6           lazyeval_0.2.2        </span></span>
<span><span class="co">## [28] cli_3.3.0              magrittr_2.0.3         crayon_1.5.1          </span></span>
<span><span class="co">## [31] poweRlaw_0.70.6        memoise_2.0.1          evaluate_0.15         </span></span>
<span><span class="co">## [34] fs_1.5.2               fansi_1.0.3            nlme_3.1-158          </span></span>
<span><span class="co">## [37] textshaping_0.3.6      tools_4.2.0            lifecycle_1.0.1       </span></span>
<span><span class="co">## [40] stringr_1.4.0          DelayedArray_0.23.0    compiler_4.2.0        </span></span>
<span><span class="co">## [43] pkgdown_2.0.5          jquerylib_0.1.4        systemfonts_1.0.4     </span></span>
<span><span class="co">## [46] rlang_1.0.4            grid_4.2.0             RCurl_1.98-1.7        </span></span>
<span><span class="co">## [49] bitops_1.0-7           rmarkdown_2.14.1       codetools_0.2-18      </span></span>
<span><span class="co">## [52] deSolve_1.32           DBI_1.1.3              R6_2.5.1              </span></span>
<span><span class="co">## [55] knitr_1.39             dplyr_1.0.9            fastmap_1.1.0         </span></span>
<span><span class="co">## [58] utf8_1.2.2             rprojroot_2.0.3        treeio_1.21.0         </span></span>
<span><span class="co">## [61] ragg_1.2.2             desc_1.4.1             ape_5.6-2             </span></span>
<span><span class="co">## [64] stringi_1.7.8          parallel_4.2.0         Rcpp_1.0.9            </span></span>
<span><span class="co">## [67] vctrs_0.4.1            tidyselect_1.1.2       xfun_0.31</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Karoline Faust, Yu Gao, Emma Gheysen, Daniel Rios Garza, Yagmur Simsek, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
